plugins/node/opentelemetry-instrumentation-router/src/utils.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as constants from './constants';
+import * as types from './types';
+import * as Router from 'router';
+
+// Detect whether a function is a router package internal plumming handler
+export const isInternal = (fn: Function) => {
+  // Note that both of those functions are sync
+  if (fn.name === 'handle' && fn.toString() === constants.ROUTER_HANDLE_FN) {
```

Comment:
why do we need to check the string body ? Isnt the name enough ?

Reply:
I'm afraid the name is too generic to be reliable in this case. Ideally, we'd reference compare the actual function called, but we do not have access to that.